CRM/Contact/BAO/Contact.php

```diff
@@ -1137,6 +1169,24 @@ public static function processImageParams(
     }
   }
 
+  /**
+   * Callback for hook_civicrm_pre().
+   * @param string $op
+   * @param string $objectName
+   * @param int $id
+   * @param array $params
+   * @throws CRM_Core_Exception
+   */
+  public static function hook_civicrm_pre($op, $objectName, $id, &$params): void {
+    // Add or update the contact image private URL in civicrm_contact
+    if (in_array($op, ['create', 'edit']) && \Civi\Api4\Utils\CoreUtil::isContact($objectName)) {
```

Comment:
Wouldn't this always be a contact cos we are on the contact object?

```suggestion
    if (in_array($op, ['create', 'edit']) ) {
```